Add a flag to override server address in the generated config

Shadi opened the pull request:
We are using Kind in one of our project integration tests suites, these tests run inside a docker container using github actions, when kind creates a cluster as part of the test run preparation it uses the default "127.0.0.1" for server address: https://github.com/kubernetes-sigs/kind/blob/main/pkg/internal/apis/config/default.go#L57

when the test runs a `kubectl` command it can't connect to the created cluster since `kubectl` is being ran from inside a container and `127.0.0.1` resolves to that container instead of the cluster on the host machine.

This change would allow us to set a different address when preparing for tests so it can execute `kubectl` commands and connect to the created cluster

BenTheElder commented:
Thank you for your PR.
In the future, per the contributor guide, we'd like to discuss features, how they will work, etc. *before* reviewing PRs.

[CONTRIBUTING.md](https://github.com/kubernetes-sigs/kind/blob/main/CONTRIBUTING.md) => https://kind.sigs.k8s.io/docs/contributing/getting-started/

We do not currently require KEPs like the rest of kubernetes, but we do prefer to flesh out the feature in an issue.

For example on such an issue I would ask:
- Does this need to be functionality in kind? It should be relatively straightforward to alter the address in kubeconfig yourself if we're just substituting it in kubeconfig vs say https://kind.sigs.k8s.io/docs/user/configuration/#api-server
- Which commands? This PR only seems to implement the flag for `kind create cluster` but there are other kubeconfig related commands
- What flag name sounds reasonable? `apiServerAddress` is not the typical naming convention (camelCase vs hypen-case) for flags and is also a misleading name IMO.
- Have we considered the related existing feature request issues and their discussions https://github.com/kubernetes-sigs/kind/issues/1558 https://github.com/kubernetes-sigs/kind/issues/2246

Shadi commented:
Hello @BenTheElder, thanks for taking a look

> Thank you for your PR. In the future, per the contributor guide, we'd like to discuss features, how they will work, etc. _before_ reviewing PRs.
> 
> [CONTRIBUTING.md](https://github.com/kubernetes-sigs/kind/blob/main/CONTRIBUTING.md) => https://kind.sigs.k8s.io/docs/contributing/getting-started/
> 
> We do not currently require KEPs like the rest of kubernetes, but we do prefer to flesh out the feature in an issue.

> For example on such an issue I would ask:
> 
> * Does this need to be functionality in kind? It should be relatively straightforward to alter the address in kubeconfig yourself if we're just substituting it in kubeconfig vs say https://kind.sigs.k8s.io/docs/user/configuration/#api-server
> * Have we considered the related existing feature request issues and their discussions [Need a way to get kubeconfig with node IP #1558](https://github.com/kubernetes-sigs/kind/issues/1558) [Improve `kind get kubeconfig` and `kind export kubeconfig` for multicluster #2246](https://github.com/kubernetes-sigs/kind/issues/2246)

this indeed similar to  [Need a way to get kubeconfig with node IP #1558](https://github.com/kubernetes-sigs/kind/issues/1558), the same `sed` workaround can be used in our case too, but it seemed it can be changed quickly by using `--name` flag as a guide so I created this PR without looking at the older issues, feel free to close this PR.

> * Which commands? This PR only seems to implement the flag for `kind create cluster` but there are other kubeconfig related commands
I'm not very familiar with this project and my goal was to only override the server address for the generated file at `KubeconfigPath`, so I didn't look at the other cases, but I'm happy to make changes for them too if you think this PR can be a starting point to implement this feature properly.

> * What flag name sounds reasonable and clearest? `apiServerAddress` is not the typical naming convention (camelCase vs hypen-case) for flags and is also a misleading name IMO.
I used `name` flag as a guide and changed it now to just `address` and updated the description.


I decided to create this PR since it seemed to be not a lot of work to implement this change in a way the `name` flag is being used now when creating clusters, if you prefer to have a proper implementation based on the discussion in  [Need a way to get kubeconfig with node IP #1558](https://github.com/kubernetes-sigs/kind/issues/1558) then feel free to close it.

aojea commented:
> `kubectl` is being ran from inside a container and `127.0.0.1` resolves to that container instead of the cluster on the host machine.

you can use `kind get kubeconfig --internal` then, it will return a kubeconfig usable inside containers

BenTheElder commented:
> you can use kind get kubeconfig --internal then, it will return a kubeconfig usable inside containers

Or better yet: I think `kind create cluster && kind export kubeconfig --internal` is what you want (`get` returns it to stdout, `export` does the kubeconfig merging like `kind create cluster`)

We should really follow up on those issues and rethink these flags for some additional options though.
